src/hooks/useSelector.js

```diff
@@ -35,16 +46,13 @@ function useSelectorWithStoreAndSubscription(
     ) {
       const newSelectedState = selector(storeState)
       // ensure latest selected state is reused so that a custom equality function can result in identical references
-      if (
-        latestSelectedState.current === undefined ||
```

Comment:
This initialization check has been moved up to the lazy initialization of `latestSelectedState`